src/Support/DataClass.php

```diff
@@ -81,6 +80,21 @@ public static function create(ReflectionClass $class): self
         );
     }
 
+    protected static function resolveAttributes(
+        ReflectionClass $class
+    ): Collection {
+        $attributes = collect($class->getAttributes())
+            ->filter(fn (ReflectionAttribute $reflectionAttribute) => class_exists($reflectionAttribute->getName()))
+            ->map(fn (ReflectionAttribute $reflectionAttribute) => $reflectionAttribute->newInstance());
+
+        $parent = $class->getParentClass();
+        if ($parent !== false && $class->getName() !== Data::class) {
```

Comment:
You want to check on the BaseData interface here, since in v4 we'll have mutliple base Data classes like Dto and Resource.

Reply:
I decided not to exclude base class provided by the package here. Easier way, not sure if we can exclude base class based on an interface it implements as the child class will also implements this interface...